src/__tests__/utils/compoundApyHelpers.test.ts

```diff
@@ -22,3 +62,61 @@ it.each([
 ])('calculate roi % with values %o', ({ amountEarned, amountInvested }, expected) => {
   expect(getRoi({ amountEarned, amountInvested })).toEqual(expected)
 })
+
+it.each`
+  apr      | compoundFrequency  | performanceFee | days   | expectedApy
+  ${68.43} | ${0}               | ${0}           | ${365} | ${0.6843}
+  ${68.43} | ${1}               | ${0}           | ${365} | ${0.9811}
+  ${68.43} | ${ONCE_PER_7_DAYS} | ${0}           | ${365} | ${0.9736}
+  ${68.43} | ${1}               | ${2}           | ${365} | ${0.9615}
+  ${68.43} | ${1}               | ${100}         | ${365} | ${0}
+  ${68.43} | ${1}               | ${0}           | ${30}  | ${0.0578}
+  ${68.43} | ${1}               | ${2}           | ${30}  | ${0.0566}
+  ${68.43} | ${1}               | ${0}           | ${1}   | ${0.00187}
+`(
+  'getApy returns $expectedApy for apr=$apr, compoundFrequency=$compoundFrequency, days=$days, performanceFee=$performanceFee',
+  ({ apr, compoundFrequency, performanceFee, days, expectedApy }) => {
+    const DIFFERENCE_THRESHOLD_IN_PERCENTS = 0.0001 // 0.01% error is OK
+    const apy = getApy(apr, compoundFrequency, days, performanceFee)
+    const difference = Math.abs(apy - expectedApy)
+    expect(difference).toBeLessThanOrEqual(DIFFERENCE_THRESHOLD_IN_PERCENTS)
+  },
+)
+
+it.each`
+  principalInUSD | apr      | earningTokenPrice | compoundFrequency   | performanceFee
+  ${84047.16}    | ${68.43} | ${17.12}          | ${1}                | ${0}
+  ${25000}       | ${68.43} | ${17.12}          | ${1}                | ${0}
+  ${25000}       | ${68.43} | ${17.12}          | ${TWICE_PER_DAY}    | ${0}
+  ${25000}       | ${68.43} | ${17.12}          | ${ONCE_PER_7_DAYS}  | ${0}
+  ${30000}       | ${68.43} | ${17.12}          | ${ONCE_PER_30_DAYS} | ${0}
+  ${30000}       | ${68.43} | ${17.12}          | ${1}                | ${2}
+  ${10000}       | ${312.5} | ${1.12}           | ${1}                | ${0}
+  ${10000}       | ${277.2} | ${0.00000012}     | ${1}                | ${0}
+  ${10000}       | ${256.1} | ${0.0012}         | ${1}                | ${0}
+  ${320000}      | ${42.0}  | ${42069.69}       | ${1}                | ${0}
+  ${320000}      | ${35.0}  | ${112000.0}       | ${1}                | ${0}
+  ${70000}       | ${112.9} | ${22.62}          | ${0}                | ${0}
+`(
+  'reverse calculations match for p=$principalInUSD, apr=$apr, earningTokenPrice=$earningTokenPrice, compoundFrequency=$compoundFrequency, performanceFee=$performanceFee',
+  ({ principalInUSD, apr, earningTokenPrice, compoundFrequency, performanceFee }) => {
+    const DIFFERENCE_THRESHOLD_IN_PERCENTS = 0.00099 // 0.099%
```

Comment:
This test ensures that Principal->ROI and ROI->Principal calculations match.
i.e. if you enter $1000 in calculator and it shows $850 as ROI for specified parameters, then if you would do the opposite - enter $850 in ROI - it will tell you that Principal to reach that ROI is $1000.
On short staking durations there might be tiny mismatches, hence the passing of test is determined by the difference between the two - 0.099%